controller/destination/server.go

```diff
@@ -104,6 +127,32 @@ func (s *server) Get(dest *common.Destination, stream pb.Destination_GetServer)
 	return nil
 }
 
+func (s *server) localKubernetesServiceIdFromDNSName(host string) (*string, error) {
+	hostLabels := strings.Split(host, ".")
+
+	// Verify that `host` ends with .svc.$zone.
+	if len(hostLabels) <= 1 + len(s.k8sDNSZoneLabels) {
```

Comment:
I think this can be more strict.  Since FQDNs must always be of the form `service.namespace.svc.$zone` then we should verify that `len(hostLabels) == 3 + len(s.k8sDNSZoneLabels)`

We can then verify that `hostLabels[2] == "svc"` and `hostLabels[3+i] == zone[i]`.